cert/cert.go

```diff
@@ -324,22 +470,43 @@ func (nc *NebulaCertificate) CheckRootConstrains(signer *NebulaCertificate) erro
 }
 
 // VerifyPrivateKey checks that the public key in the Nebula certificate and a supplied private key match
-func (nc *NebulaCertificate) VerifyPrivateKey(key []byte) error {
+func (nc *NebulaCertificate) VerifyPrivateKey(curve Curve, key []byte) error {
```

Comment:
Could we use `nc.Details.Curve` instead of passing the curve in here?

Reply:
Yeah I think you are correct we could do that. I think the intent was to verify the curve defined in the private key file matches the curve in the certificate, but I don't think this code is doing that (directly at least). I'll take a closer look.

Reply:
I pushed a change to do the original intent, which is to verify the curves in the private key and certificate match: https://github.com/slackhq/nebula/pull/769/commits/edc283d27a54193d74168a72f054fbf5b5bf21c6